solutions/09-11/script.js

```diff
@@ -0,0 +1,97 @@
+function validateCardNumber(value) {
+    return !(/[^0-9-\s]+/.test(value));
+}
+
+function luhnAlghorithm(value) {
+    let nCheck = 0, bEven = false;
+	value = value.replace(/\D/g, "");
+
+	for (let n = value.length - 1; n >= 0; n--) {
+		let cDigit = value.charAt(n),
+			  nDigit = parseInt(cDigit, 10);
+
+		if (bEven && (nDigit *= 2) > 9) nDigit -= 9;
+
+		nCheck += nDigit;
+		bEven = !bEven;
+	}
+	return (nCheck % 10) == 0;
+}
+
+// Card recognition function - first char is reserved for card type
+function whatCard(value) {
+    switch(value[0]) {
+        case '3': {
```

Comment:
I don't know what means that cases, this number can make problems, maybe better make map object:
```

const typeCards = {
visa: 'visa',
mastercard: 'mastercard;
}
```

Reply:
Good point, cardTypes object added